src/main/viz/GeneTrack.js

```diff
@@ -144,23 +144,25 @@ class GeneTrack extends React.Component {
                                 clampedScale(1 + gene.position.stop()), geneLineY + 0.5);
 
       // TODO: only compute all these intervals when data becomes available.
-      var exons = bedtools.splitCodingExons(gene.exons, gene.codingRegion);
-      exons.forEach(exon => {
-        ctx.fillRect(sc(1 + exon.start),
-                     geneLineY - 3 * (exon.isCoding ? 2 : 1),
-                     sc(exon.stop + 2) - sc(1 + exon.start),
-                     6 * (exon.isCoding ? 2 : 1));
-      });
-
-      var introns = gene.position.interval.complementIntervals(gene.exons);
-      introns.forEach(range => {
-        drawArrow(ctx, clampedScale, range, geneLineY + 0.5, gene.strand);
-      });
-      ctx.strokeStyle = style.GENE_COMPLEMENT_COLOR;
-      ctx.lineWidth = 2;
-      gene.exons.forEach(range => {
-        drawArrow(ctx, clampedScale, range, geneLineY + 0.5, gene.strand);
-      });
+      if (gene.strand && gene.exons && gene.codingRegion) {
```

Comment:
I don't think we want to modify the GeneTrack code. If strand and exons and codingRegion are missing, FeatureTrack would be a better fit for the data.

Reply:
I agree that a FeatureTrack is a better fit for the data, but I don't know which data is expected for each visualization type.   I don't know when I'm supposed to use a FeatureTrack vs a CoverageTrack vs a GenomeTrack vs a GeneTrack when I'm a naive user with a BAM or BED file in hand.

I try to utilize defensive programming for user naivety and make sure something runs and doesn't throw errors if the user attempts to do something not intended by the code.  Wrapping this set of code in an if statement makes it so that the GeneTrack loads even when provided a simple BED file with 3 columns.

Without these changes, GeneTrack throws:
```
bedtools.js:21 Uncaught TypeError: Cannot read property 'map' of undefined
    at Object.splitCodingExons (bedtools.js:21)
    at GeneTrack.js:147
    at Array.forEach (<anonymous>)
    at GeneTrack.updateVisualization (GeneTrack.js:136)
    at GeneTrack.componentDidUpdate (GeneTrack.js:104)
    at CallbackQueue.notifyAll (CallbackQueue.js:65)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:81)
    at ReactReconcileTransaction.closeAll (Transaction.js:202)
    at ReactReconcileTransaction.perform (Transaction.js:149)
    at ReactUpdatesFlushTransaction.perform (Transaction.js:136)
```